content/en/templates/data-templates.md

```diff
@@ -28,11 +28,15 @@ Hugo supports loading data from YAML, JSON, XML, and TOML files located in the `
 
 The `data` folder is where you can store additional data for Hugo to use when generating your site. Data files aren't used to generate standalone pages; rather, they're meant to be supplemental to content files. This feature can extend the content in case your front matter fields grow out of control. Or perhaps you want to show a larger dataset in a template (see example below). In both cases, it's a good idea to outsource the data in their own files.
 
-These files must be YAML, JSON, XML, or TOML files (using the `.yml`, `.yaml`, `.json`, `.xml`, or `.toml` extension). The data will be accessible as a `map` in the `.Site.Data` variable.
+These files must be YAML, JSON, XML, or TOML files (using the `.yml`, `.yaml`, `.json`, `.xml`, or `.toml` extension). The data will be accessible as a `map` in the `.Site.Data` variable. Additionally, filenames must begin with an underscore or a Unicode letter, followed by zero or more underscores, Unicode letters, or Unicode digits. eg:
+
+- `2022.json` - Invalid
+- `data_2022.json` - Valid
+- `_2022.json` - Valid
```

Comment:
```suggestion
- `_123.json` (valid)

If you wish to access the data using the [`index`](/functions/index-function/) function, the first character of the filename is irrelevant. For example:

Date file|Template code
:--|:--
`123.json`|`{{ index .Site.Data "123" }}`
`x123.json`|`{{ index .Site.Data "x123" }}`
`_123.json`|`{{ index .Site.Data "_123" }}`
```

Reply:
I think line 41 here is a mistake?  123.json is an invalid filename.

Reply:
> 123.json is an invalid filename.

Not when accessed with the index function.

Reply:
I pushed the changes for this thought slightly modified (you used "(valid)" where the previous lines used "- Valid". I changed it to match the previous lines (though I have no preference and could change them all to use "(valid)" if you prefer